daft/io/lance/lance_scan.py

```diff
@@ -0,0 +1,101 @@
+# ruff: noqa: I002
+# isort: dont-add-import: from __future__ import annotations
+
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Optional
+
+from daft.daft import PyPartitionField, PyPushdowns, PyRecordBatch, ScanTask
+from daft.io.scan import ScanOperator
+from daft.logical.schema import Schema
+from daft.recordbatch import RecordBatch
+
+if TYPE_CHECKING:
+    import lance
+    import pyarrow
+
+
+# TODO support fts and fast_search
+def _lancedb_table_factory_function(
+    ds: "lance.LanceDataset",
+    fragment_ids: Optional[list[int]] = None,
+    required_columns: Optional[list[str]] = None,
+    filter: Optional["pyarrow.compute.Expression"] = None,
+    limit: Optional[int] = None,
+    use_scalar_index: bool = False,
+) -> Iterator[PyRecordBatch]:
+    fragments = [ds.get_fragment(id) for id in (fragment_ids or [])]
+    assert fragments, RuntimeError(f"Unable to find lance fragments {fragment_ids}")
```

Comment:
This would still return an `AssertionError` with a `RuntimeError` inside of it. Can we rewrite this to:

```py
if not fragments:
    raise RuntimeError(f"Unable to find lance fragments {fragment_ids}")
```